### PR TITLE
Watch for changes on HYP_CONTENT_FOR_IDS

### LINKED ISSUE #2: Change yielded content on route changes

### LINKED ISSUE #3: Watch HYP_CONTENT_FOR_IDS for changes
Removing the usage of the _$interval_ service and replacing it with some sort of watch functionality would improve the speed and performance of the application. It would probably make the code more readable as well. It would also probably fix #2.

### PR DESCRIPTION
**Pull request by sindrenm**
Fixes #2.
Fixes #3.

### CODE REVIEW

**Comment by sindrenm** [comment]
Thanks to @marcel-eggum for putting me on the right track. :smiley:

File: hyp-content-for.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
 angular.module("hypContentFor", [])
-  .constant("HYP_CONTENT_FOR_IDS", [ ])
+  .value("HYP_CONTENT_FOR_IDS", { })
```
Comment (mathiasbno) [inline hyp-content-for.js:2]: Liking this! :+1:

File: hyp-content-for.js [hunk 2]
```diff
@@ -20,20 +20,12 @@ angular.module("hypContentFor", [])
...
-
-          elem.replaceWith(content);
-        }
-
-        repeatFn();
-        interval = $interval(repeatFn, 100, 9);
+        scope.$watch(watchFn, function (newValue) {
+          elem.empty();
+          elem.append(newValue);
+        });
```
Comment (mathiasbno) [inline hyp-content-for.js:28]: So much better :)

File: hyp-content-for.js (hunk 1 above)
Comment (sindrenm) [inline hyp-content-for.js:2]: Yeah, it kinda made more sense as the content of the collection would be changing. And the array thing was just wrong, it was supposed to be an object from the get-go. :smiley:

File: README.md [hunk 3]
```diff
@@ -50,7 +50,9 @@ _Resulting layout:_
 ```html
 <header>
   <h1>Blog 2000</h1>
-  <h2>All articles</h2>
+  <yield to="header">
```
Comment (mathiasbno) [inline README.md:53]: So I know that this is influenced by rails, but is `to` the right word here? In my head `target` or something in those lines make more sense.

File: hyp-content-for.js (hunk 2 above)
Comment (sindrenm) [inline hyp-content-for.js:28]: Yes! :grinning:

File: README.md (hunk 3 above)
Comment (sindrenm) [inline README.md:53]: Hmm, I wasn't really sure what to call that attribute, to be honest. `target` could work, I suppose. However, I feel like it should read as if it was said out loud. “Yield to toolbar” or “yield target toolbar” are both okay-ish. Or maybe even `yield` is wrong. As you say, that's heavily influenced by Rails, but we're at liberty to call it whatever we want. :smiley:

Comment (mathiasbno) [inline README.md:53]: We can refactor this later. Maybe it should be `<pop in="header">` for all we know, reads good ^^

**Pull request merged**